### PR TITLE
Random cleanup

### PR DESCRIPTION
**Pull request by jonsterling**
This is up to date with #63; I will wait until that is merged.

### CODE REVIEW

File: src/lib/grammar.mly [hunk 1]
```diff
@@ -77,10 +77,10 @@ plain_atomic_in_cof: t = plain_atomic_term | t = plain_atomic_cof_except_var {t}
 plain_cof_except_var:
   | c = plain_atomic_cof_except_var
     { c }
-  | phi = located(plain_atomic_in_cof) JOIN psi = located(plain_atomic_in_cof)
-    { Join (phi, psi) }
-  | phi = located(plain_atomic_in_cof) MEET psi = located(plain_atomic_in_cof)
-    { Meet (phi, psi) }
+  | phi = located(plain_atomic_in_cof) JOIN psis = separated_nonempty_list(JOIN, located(plain_atomic_in_cof))
```
Comment (favonia) [inline src/lib/grammar.mly:80]: The lost Ch'i from #70 was recovered.

**Comment by favonia** [comment]
The code seems to be shorter (?) with n-ary joins and meets.

File: src/lib/Monads.ml [hunk 2]
```diff
@@ -154,15 +155,14 @@ struct
       abort_if_inconsistent () @@
       match cofs with
       | [] -> m
-      | (Cof.Var _ | Cof.Cof (Cof.Top | Cof.Bot | Cof.Eq _)) as phi :: cofs ->
+      | (Cof.Var _ | Cof.Cof (Cof.Meet [] | Cof.Join [] | Cof.Eq _)) as phi :: cofs ->
```
Comment (favonia) [inline src/lib/Monads.ml:158]: Question: do we need special cases for `Cof.Meet []` and `Cof.Join []` here?

Comment (jonsterling) [reply src/lib/Monads.ml:158]: Yes.

Comment (favonia) [reply src/lib/Monads.ml:158]: Are you sure? `restrict (Cof.Meet [])` seems to be doing nothing. Not sure about `restrict (Cof.Join [])`, though.

**Pull request merged**